### DISCUSSION TITLE
Non Authenticated user session, How to handle this in Supabase to track and keep sessions for those users?

### DISCUSSION THREAD

**Discussion by juangbiestro** [supabase/supabase#22254] [Questions] [answered]
Hi,
I would like to know how I can handle session and policy access for visitors or non-authenticated users in my web app using Supabase. I could not find anything related to non-authenticated users to return or generate any sort of JWT or hash with the token or session for those users. Is there any solution or documentation available to handle this? Cheers.

**Comment by GaryAustin1** [comment]
There is work underway to add anonymous sign in.  https://github.com/supabase/auth/pull/1460

Some one asking about status: https://github.com/orgs/supabase/discussions/22216

Otherwise some of have used fake emails to create users.  
You can also mint your own jwt's with some info in the claims or a fake UUID just for RLS.

**Reply by juangbiestro** [reply]
is there any example for this in JS using React or Next.js ? cheers.

**Reply by GaryAustin1** [reply]
Minting your own jwt's would involve getting the claims correctly https://supabase.com/docs/guides/auth/auth-deep-dive/auth-deep-dive-jwts#jwts-in-supabase and then using a jwt library server side as you have to use the jwt secret.

I've only seen users discuss using fake email in Discord, but not seen a complete solution.

**Reply by GaryAustin1** [reply, marked as answer]
https://supabase.com/docs/guides/auth/auth-anonymous
Anonymous users are now available.

**Reply by juangbiestro** [reply]
Thanks Gary
